### ISSUE TITLE
Cannot build nextjs app on FreeBSD

### ISSUE THREAD

**Issue by patrykluba** [vercel/next.js#59997]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: freebsd
  Arch: x64
  Version: FreeBSD 13.2-RELEASE #0 releng/13.2-n254617-525ecfdad597-dirty: Wed May 31 19:03:43 CEST 2023     
Binaries:
  Node: 20.10.0
  npm: 9.6.6
  Yarn: 1.22.19
  pnpm: N/A
Relevant Packages:
  next: 14.0.4 / 14.0.5-canary.29
  eslint-config-next: 14.0.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which example does this report relate to?

hello-world

### How are you deploying your application? (if relevant)

yarn build / yarn start

### Describe the Bug

I have error:

```
[user@s62]:<~/domains/test.net/my-app>$ yarn build
yarn run v1.22.19
$ next build
   ▲ Next.js 14.0.4

   Downloading swc package @next/swc-freebsd-x64...
 ⨯ Failed to download swc package from https://registry.yarnpkg.com/@next/swc-freebsd-x64/-/swc-freebsd-x64-14.0.4.tgz
unhandledRejection Error: request failed with status 404
```
Even I adding to .babelrc:

```
{
  "presets": ["next/babel"]
}
```
and swcMinifty: false to next.config.js build still failing

### Expected Behavior

App should build

### To Reproduce

yarn build

**Comment by ismatim** [comment]
![pic-full-240104-0929-33](https://github.com/vercel/next.js/assets/4640195/3e96b929-efad-497e-83f6-47353b5c0fae)

It happens on OpenBS 7.4 and in FreeBSD 14.

**Comment by Vringe** [comment]
Same here with FreeBSD 13.2

**Comment by wangii** [comment]
I had the same problem, even if I built the next-swc with rust it still crashed. I managed to get it running by using @next/swc-wasm-nodejs, and then change the `node_modules/next/dist/build/swc/index.js`, line 247, `shouldLoadWasmFallbackFirst` to `true`.

hope it helps.

**Comment by tinola** [comment]
@wangii, which FreeBSD version do you use? I tried to build with your suggestion, but with no luck.

```
> next build

   Skipping linting
 ✓ Checking validity of types
   ▲ Next.js 14.0.3
   - Environments: .env
   - Experiments (use at your own risk):
     · instrumentationHook

   Downloading swc package @next/swc-wasm-nodejs...
 ⨯ Failed to download swc package from https://registry.npmjs.org/@next/swc-wasm-nodejs/-/swc-wasm-nodejs-14.0.3.tgz
 ⚠ Attempted to load @next/swc-freebsd-x64, but it was not installed
 ⨯ Failed to load SWC binary for freebsd/x64, see more info here: https://nextjs.org/docs/messages/failed-loading-swc

```

**Comment by wangii** [comment]
The FreeBSD version I'm using: 14.0-RELEASE.

@tinola I don't think it's the version of FreeBSD matters here, it's the next.js version. I'm using 14.1.0. could you try this version of next?

**Comment by tinola** [comment]
You're right, OS version doesn't matter.
After upgrade next to v.14.1.0, building app went just fine.
Thanks!

**Comment by EmdevX** [comment]
This issue still persist in the latest canary version. I would rather use experimental option instead of editing node_modules itself, but `useWasmBinary` is not working as expected, because it is not passed in some of the `loadBindings` function instances

https://github.com/vercel/next.js/blob/canary/packages/next/src/build/swc/index.ts#L1478
https://github.com/vercel/next.js/blob/canary/packages/next/src/build/swc/index.ts#L1473
https://github.com/vercel/next.js/blob/canary/packages/next/src/build/swc/index.ts#L1468
https://github.com/vercel/next.js/blob/canary/packages/next/src/build/swc/index.ts#L1463
https://github.com/vercel/next.js/blob/canary/packages/next/src/build/swc/index.ts#L1458
https://github.com/vercel/next.js/blob/canary/packages/next/src/build/swc/index.ts#L1453

**Comment by avshyz** [comment]
Seems related to this
https://github.com/vercel/next.js/issues/58483

I receive a similar error when I try to build my next project on a raspberry pi

```
⨯ Failed to download swc package from https://registry.npmjs.org/@next/swc-linux-arm-gnueabihf/-/swc-linux-arm-gnueabihf-14.1.0.tgz
```

**Comment by davidnbooth** [comment]
I have the same issue.  FreeBSD 13.2-RELEASE

**Comment by colonder** [comment]
@ijjk

**Comment by colonder** [comment]
@kwonoj

**Comment by colonder** [comment]
Yeah, so they deprecated freebsd as well as linux arm https://github.com/vercel/next.js/blob/4efe14238b5ab11935e73aa09631ef5ec8045b13/packages/next/src/build/swc/index.ts#L52

**Comment by Vringe** [comment]
Can we please get an official statement if this is going to be fixed or if FreeBSD support actually is going to be dropped?

**Comment by erkisun** [comment]
I wonder if freebsd and nextjs will have a future ... is there anything left ? 

▲ Next.js 14.2.5
  - Local:        http://localhost:3000

 ✓ Starting...
Error: Could not find a production build in the '.next' directory. .....

**Comment by Vringe** [comment]
Unfortunately, the maintainers of SWC are not interested in maintaining a build for FreeBSD anymore.

https://github.com/swc-project/swc/issues/10106

**Comment by marcoshernanz** [comment]
Thanks for reporting this. The missing native FreeBSD package is expected because native bindings are no longer published for FreeBSD. The WASM fallback regressions were fixed in #64567 and #65633. Current Next.js supports FreeBSD through WASM with webpack, using `next dev --webpack` or `next build --webpack`, so I’m closing this issue as resolved.
